src/Microsoft.AspNetCore.Mvc.Formatters.Xml/ProblemDetailsWrapper.cs

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Globalization;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+
+namespace Microsoft.AspNetCore.Mvc.Formatters.Xml
+{
+    /// <summary>
+    /// Wrapper class for <see cref="Mvc.ProblemDetails"/> to enable it to be serialized by the xml formatters.
+    /// </summary>
+    [XmlRoot(nameof(ProblemDetails))]
+    public sealed class ProblemDetailsWrapper : IXmlSerializable, IUnwrappable
+    {
+        /// <summary>
+        /// Initializes a new instance of <see cref="ProblemDetailsWrapper"/>.
+        /// </summary>
+        public ProblemDetailsWrapper()
+            : this(new ProblemDetails())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of <see cref="ProblemDetailsWrapper"/>.
+        /// </summary>
+        public ProblemDetailsWrapper(ProblemDetails problemDetails)
+        {
+            ProblemDetails = problemDetails;
+        }
+
+        /// <summary>
+        /// Gets the wrapped <see cref="Mvc.ProblemDetails"/>.
+        /// </summary>
+        public ProblemDetails ProblemDetails { get; }
+
+        /// <inheritdoc />
+        public XmlSchema GetSchema() => null;
+
+        /// <inheritdoc />
+        public void ReadXml(XmlReader reader)
+        {
+            if (reader.IsEmptyElement)
+            {
+                reader.Read();
+                return;
+            }
+
+            reader.ReadStartElement();
+            while (reader.NodeType != XmlNodeType.EndElement)
+            {
+                var key = XmlConvert.DecodeName(reader.LocalName);
+                ReadProperty(reader, ProblemDetails, key);
+
+                reader.MoveToContent();
+            }
+
+            reader.ReadEndElement();
+        }
+
+        internal static void ReadProperty(XmlReader reader, ProblemDetails problemDetails, string key)
+        {
+            var value = reader.ReadInnerXml();
+
+            if (key == nameof(problemDetails.Detail))
+            {
+                problemDetails.Detail = value;
+            }
+            else if (key == nameof(problemDetails.Instance))
+            {
+                problemDetails.Instance = value;
+            }
+            else if (key == nameof(problemDetails.Status))
+            {
+                problemDetails.Status = string.IsNullOrEmpty(value) ?
+                    (int?)null :
+                    int.Parse(value, CultureInfo.InvariantCulture);
+            }
+            else if (key == nameof(problemDetails.Title))
+            {
+                problemDetails.Title = value;
+            }
+            else if (key == nameof(problemDetails.Type))
+            {
+                problemDetails.Type = value;
+            }
+            else
```

Comment:
Of course, would need to special case `Errors` here if `ValidationProblemDetailsWrapper` were a subclass.